src/main/java/com/infobip/jira/JiraService.java

```diff
@@ -15,30 +15,139 @@
+		}
+	}
+
+	void releaseVersion(SerializedVersion version, LocalDate releaseDate) {
+
+		String body = String.format(
+				"{\"released\":true,\"releaseDate\":\"%s\"}",
+				releaseDate);
+
+		executeJsonHttpRequest(MethodType.PUT, "/rest/api/2/version/" + version.getId(), body);
+	}
+
+	private List<SerializedVersion> getVersions(ProjectKey projectKey) {
+
+		String url = getUrl("/rest/api/2/project/", projectKey.getValue(), "/versions");
+		String versionsJson = executeJsonHttpRequest(MethodType.GET, url);
+
+		try {
+			return objectMapper.readValue(versionsJson, new TypeReference<List<SerializedVersion>>() {});
+		} catch (IOException e) {
+			throw new JiraServiceException("Failed ", e);
+		}
+	}
+
+	private ApplicationLinkRequestFactory getJiraApplicationLinkRequestFactory() {
+
+		ApplicationLink applicationLink = applicationLinkService.getPrimaryApplicationLink(JiraApplicationType.class);
+
+		if (applicationLink == null) {
+			throw new IllegalStateException("Primary JIRA application link does not exist!");
+		}
+
+		return applicationLink.createAuthenticatedRequestFactory();
+	}
+
+	private String executeJsonHttpRequest(MethodType methodType, String url, Object body) {
+
+		return executeJsonHttpRequest(methodType, url, toJson(body));
+	}
+
+	private String executeJsonHttpRequest(MethodType methodType, String url, String body) {
+
+		try {
+			return getJiraApplicationLinkRequestFactory().createRequest(methodType, url)
+					.setHeader("Content-Type", "application/json")
+					.setRequestBody(body).execute();
+		} catch (CredentialsRequiredException | ResponseException e) {
+			throw new JiraServiceException("Failed to create request " + methodType + " " + url + " " + body, e);
+		}
+	}
+
+	private String executeJsonHttpRequest(MethodType methodType, String url) {
 
-	boolean doesJiraVersionExist(Version version) throws CredentialsRequiredException, ResponseException, IOException;
+		try {
+			return getJiraApplicationLinkRequestFactory().createRequest(methodType, url)
+					.setHeader("Content-Type", "application/json").execute();
+		} catch (CredentialsRequiredException | ResponseException e) {
+			throw new JiraServiceException("Failed to create request " + methodType + " " + url, e);
+		}
+	}
 
-	void createJiraVersion(Version version) throws
-	                                        CredentialsRequiredException,
-	                                        ResponseException,
-	                                        JsonProcessingException;
+	private String getUrl(String... parts) {
 
-	void addVersionToIssues(Version version, List<IssueKey> issueKeys) throws
-	                                                                   CredentialsRequiredException,
-	                                                                   ResponseException,
-	                                                                   JsonProcessingException;
+		return Joiner.on("").join(parts);
```

Comment:
inlined the helper method and replaced with String.format as it is more readable